bins/reth-crawler/src/crawler/listener/update_listener.rs

```diff
@@ -169,6 +180,27 @@ impl UpdateListener {
                     let best_block = their_status.blockhash.to_string();
                     let genesis_block_hash = their_status.genesis.to_string();
 
+                    // check if peer is synced with the latest chain's blocks
+                    let mut synced = None;
+                    if let Ok(last_block_number) = provider.get_block_number().await {
+                        info!("last block number: {}", last_block_number);
+                        let peer_best_block_hash = Into::<H256>::into(their_status.blockhash.0);
+                        info!("peer best block hash: {:#?}", peer_best_block_hash);
+                        if let Ok(Some(peer_best_block)) =
+                            provider.get_block(peer_best_block_hash).await
+                        {
+                            let peer_best_block_number =
+                                peer_best_block.number.expect("it's not a pending block!");
+                            if peer_best_block_number < last_block_number - SYNCED_THRESHOLD {
+                                synced = Some(false);
+                            } else {
+                                synced = Some(true);
+                            }
+                            info!("peer best block number: {}", peer_best_block_number);
+                            info!("sync: {}", synced.unwrap());
+                        }
+                    }
```

Comment:
id make an issue to get rid of the duplication between the code here and the code in `start_dnsdisc`

Reply:
yes. On this point I just copy paste here because we can later refactor all the `update_listener.rs` file since a there is a lot of duplication in that file.